src/main/drivers/pwm_output.c

```diff
@@ -380,3 +385,41 @@ void servoDevInit(const servoDevConfig_t *servoConfig)
 }
 
 #endif
+
+
+#ifdef BEEPER
+void pwmWriteBeeper(bool onoffBeep)
+{
+        if(onoffBeep == true){
+            *beeperPwm.ccr = (1000000/freqBeep)/2;
+            beeperPwm.enabled = true;
+        } else {
+            *beeperPwm.ccr = 0;
+            beeperPwm.enabled = false;
+        }
+}
+
+void pwmToggleBeeper(void)
+{
+        pwmWriteBeeper(!beeperPwm.enabled);
+}
+
+void beeperPwmInit(uint16_t frequency)
```

Comment:
Good point. With the code as it is now, what happens if `frequency` is != 0, and `BEEPER` has no `IO_t` or no timer (and hence is not initialised), and then `pwmWriteBeeper` is called?